examples/computecloth/computecloth.cpp

```diff
@@ -694,23 +737,47 @@ class VulkanExample : public VulkanExampleBase
 
 	void draw()
 	{
-	VkSubmitInfo computeSubmitInfo = vks::initializers::submitInfo();
-	computeSubmitInfo.commandBufferCount = 1;
-	computeSubmitInfo.pCommandBuffers = &compute.commandBuffers[readSet];
-
-	VK_CHECK_RESULT( vkQueueSubmit( compute.queue, 1, &computeSubmitInfo, compute.fence ) );
+        static bool firstDraw = true;
+        VkSubmitInfo computeSubmitInfo = vks::initializers::submitInfo();
+        // FIXME find a better way to do this (without using fences, which is much slower)
+        VkPipelineStageFlags computeWaitDstStageMask = VK_PIPELINE_STAGE_VERTEX_INPUT_BIT;
+        if (!firstDraw) {
+            computeSubmitInfo.waitSemaphoreCount = 1;
+            computeSubmitInfo.pWaitSemaphores = &compute.semaphores.ready;
+            computeSubmitInfo.pWaitDstStageMask = &computeWaitDstStageMask;
+        } else {
+            firstDraw = false;
+        }
+        computeSubmitInfo.signalSemaphoreCount = 1;
+        computeSubmitInfo.pSignalSemaphores = &compute.semaphores.complete;
+	    computeSubmitInfo.commandBufferCount = 1;
+	    computeSubmitInfo.pCommandBuffers = &compute.commandBuffers[readSet];
+
+	    VK_CHECK_RESULT( vkQueueSubmit( compute.queue, 1, &computeSubmitInfo, VK_NULL_HANDLE) );
 
 		// Submit graphics commands
 		VulkanExampleBase::prepareFrame();
 
+        VkPipelineStageFlags waitDstStageMask[2] = {
+            submitPipelineStages, VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT
```

Comment:
`BOTTOM_OF_PIPE_BIT` ->`VERTEX_INPUT_BIT` (wait on compute being done before reading the vertex data).

Reply:
thanks, fixed

Reply:
Does this not also imply that https://github.com/SaschaWillems/Vulkan/blob/master/examples/computecullandlod/computecullandlod.cpp#L806 is incorrect?  In other words the `VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT` isn't likely to ever be a correct destination wait stage for a command buffer containing only graphics pipeline operations?

Reply:
Yeah, agree that seems wrong as well.